### PR TITLE
feat: grpc handler result

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add GREPTIME_ERR_CODE/GREPTIME_ERR_MSG to gRPC header

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by fengjiachun** [comment]
@killme2008 @shuiyisong PTAL

File: src/servers/src/error.rs [hunk 1]
```diff
@@ -463,11 +463,11 @@ impl From<Error> for tonic::Status {
         // (which is a very rare case), just ignore. Client will use Tonic status code and message.
         let status_code = err.status_code();
         if let Ok(code) = HeaderValue::from_bytes(status_code.to_string().as_bytes()) {
-            let _ = headers.insert(INNER_ERROR_CODE, code);
+            let _ = headers.insert(GREPTIME_ERROR_CODE, code);
```
Comment (killme2008) [inline src/servers/src/error.rs:466]: Looks like the code is a string just them in https://greptimedb.rs/common_error/status_code/enum.StatusCode.html

Prefer to return the integer code instead.

https://greptimedb.rs/src/common_error/status_code.rs.html#21-88

The integer code is unchanged for ever, but the name may be changed. And integer code is more friendly for error processing.

**Comment by fengjiachun** [comment]
@killme2008 PTAL

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by shuiyisong — APPROVED** [review]
LGTM

**Pull request merged**